Improve CI configuration

martin-georgiev opened the pull request.

martin-georgiev committed:
Prepare bundle for Symfony 4 compatibility

martin-georgiev committed:
Updated readme and added license node to composer.json

martin-georgiev committed:
Fix tests (add parameter value for kernel's directory name)

martin-georgiev committed:
Drop unnecessary compiler pass

martin-georgiev committed:
CI tweaks

martin-georgiev committed:
Merge branch 'master' into feature/symfony4-ready

ben-challis commented on README.md:
```diff
@@ -81,7 +81,7 @@ lendable_doctrine_extensions:
         App\Entity\Repository\ExampleRepository:
             entity: App\Entity\Example
-            managers: [default]
+            managers: [your_default_manager]
```
@martin-georgiev this refers to the "default" manager exposed by DoctrineBundle, maybe best to write this as `['default', 'custom_manager']` perhaps?

martin-georgiev committed:
Require specific minor PHP version
(security-checker suffers from https://github.com/travis-ci/travis-ci/issues/6339#issuecomment-355158125)

martin-georgiev committed:
Merge branch 'feature/symfony4-ready' of https://github.com/Lendable/doctrine-extensions-bundle into feature/symfony4-ready

# Conflicts:
#	.travis.yml

martin-georgiev committed:
Do not check security for PHP 7.0 (Travis bug)

ben-challis commented on .travis.yml:
```diff
@@ -6,45 +6,62 @@ cache:
-  - composer self-update
-  # Set composer minimum-stability configuration filter to beta versions
-  - if [ "$DEPENDENCIES" = "beta" ]; then perl -pi -e 's/^}$/,"minimum-stability":"beta"}/' composer.json; fi;
+  - composer install --no-interaction
```
@martin-georgiev this should move to the end of the install definitions, the `composer config minimum-stability` and `composer require` at present won't work (separate comment regarding `require`).

ben-challis replied:
@martin-georgiev this should probably also be `composer update -n --prefer-dist` again as no lock file

ben-challis commented on .travis.yml:
```diff
@@ -6,45 +6,62 @@ cache:
-  - if [ "$deps" = "no" ]; then composer update; fi;
-  - if [ "$deps" = "low" ]; then composer --prefer-lowest --prefer-stable update; fi;
+  - if [ "$DEPENDENCIES" = "beta" ]; then composer config minimum-stability beta; fi;
+  - if [ "$SYMFONY_VERSION" != "" ]; then composer require --no-update symfony/symfony:${SYMFONY_VERSION}; fi
```
@martin-georgiev there's no lock file so `--no-update` here is not required.

martin-georgiev committed:
Tweaks to Travis install steps
CR feedback

martin-georgiev committed:
Update min version required for php-cs-fixer and coveralls

martin-georgiev committed:
Do not run unnecessary file validation and code style checks for matrices with older symfony version

martin-georgiev committed:
Travis to drop friendsofphp/php-cs-fixer as dev dependencies when symfony not latest

martin-georgiev committed:
Improve clarity of env values with quotes

martin-georgiev committed:
Support earlier versions of symfony/yaml

martin-georgiev committed:
Support earlier versions of symfony/yaml

martin-georgiev committed:
Force remove composer.lock that may get created for some matrices

martin-georgiev committed:
Remove symfony/yaml requirement as it is covered by symfony/symfony

martin-georgiev committed:
Downgrade requirement for doctrine/orm to ^2.4

martin-georgiev committed:
Reformat long lines

martin-georgiev committed:
Wrap variable usages in quotes to ensure the shell isn't messing up

martin-georgiev committed:
Fix bug where Travis didn't require symfony as dev dependency

martin-georgiev committed:
Add description node to composer.json

martin-georgiev committed:
Update Symfony with dependencies

martin-georgiev committed:
Tweaks in composer steps

martin-georgiev committed:
Run security check only for maintained Symfony versions
